openpype/hosts/maya/api/lib.py

```diff
@@ -2610,69 +2669,32 @@ def _get_attrs(node):
-                                 new_node_attr,
-                                 force=True)
-
-            # If shapes are also baked then connect those keyable attributes
-            if shape:
-                children_shapes = cmds.listRelatives(new_node,
-                                                     children=True,
-                                                     fullPath=True,
-                                                     shapes=True)
-                if children_shapes:
-                    orig_children_shapes = cmds.listRelatives(node,
-                                                              children=True,
-                                                              fullPath=True,
-                                                              shapes=True)
-                    for orig_shape, new_shape in zip(orig_children_shapes,
-                                                     children_shapes):
-                        attrs = _get_attrs(orig_shape)
-                        for attr in attrs:
-                            orig_node_attr = '{0}.{1}'.format(orig_shape, attr)
-                            new_node_attr = '{0}.{1}'.format(new_shape, attr)
-
-                            # unlock to avoid connection errors
-                            cmds.setAttr(new_node_attr, lock=False)
-
-                            cmds.connectAttr(orig_node_attr,
-                                             new_node_attr,
-                                             force=True)
-
-            # Parent to world
-            if cmds.listRelatives(new_node, parent=True):
-                new_node = cmds.parent(new_node, world=True)[0]
-
-            # Unlock transform attributes so constraint can be created
-            for attr in transform_attrs:
-                cmds.setAttr('{0}.{1}'.format(new_node, attr), lock=False)
-
-            # Constraints
-            delete_bin.extend(cmds.parentConstraint(node, new_node, mo=False))
-            delete_bin.extend(cmds.scaleConstraint(node, new_node, mo=False))
-
-            world_space_nodes.append(new_node)
-
-        bake(world_space_nodes,
-             frame_range=frame_range,
-             step=step,
-             simulation=simulation,
-             preserve_outside_keys=preserve_outside_keys,
-             disable_implicit_control=disable_implicit_control,
-             shape=shape)
+            with _revert_lock_attributes(node, new_node):
+                with _revert_lock_shape_attributes(node, new_node):
+                    # Parent to world
+                    if cmds.listRelatives(new_node, parent=True):
+                        new_node = cmds.parent(new_node, world=True)[0]
+
+                    # Unlock transform attributes so constraint can be created
+                    for attr in transform_attrs:
+                        cmds.setAttr(
+                            '{0}.{1}'.format(new_node, attr), lock=False)
```

Comment:
Do we still need to do this if we're already passing through context managers dealing with locking/unlocking these attributes. These transform attributes, if locked before should also remain locked after and thus should be handled in the context managers. Right?